api/controllers/link-controller.ts

```diff
@@ -29,7 +34,19 @@ export const createLink = async (
     })
     .toArray();
 
-  if (result.length !== 0) return createLink(longUrl, ipAddress);
+  if (result.length !== 0) {
+    if (!customCode) return createLink(longUrl, ipAddress);
+    else {
+       result = await database
```

Comment:
I understand why let was used, but why are you querying the db again? Attach a ss here to justify it.

Reply:
The second time is to check whether the link was not created due to `customCode` existing already.